InformationScripting/src/commands/CScript.cpp

```diff
@@ -88,6 +98,69 @@ Interaction::CommandResult* CScript::execute(Visualization::Item*, Visualization
 			qDebug() << "Error in Python: " << BoostPythonHelpers::parsePythonException();
 		}
 	}
+	else if (command == "methods")
+	{
+		auto query = AstSource::instance().createMethodQuery(node, args);
+		auto compositeQuery = new CompositeQuery();
+		compositeQuery->connectToOutput(query);
+		QueryExecutor queryExecutor(compositeQuery);
+		queryExecutor.execute();
+	}
+	else if (command == "bases")
+	{
+		auto query = AstSource::instance().createBaseClassesQuery(node, args);
+		QueryExecutor queryExecutor(query);
+		queryExecutor.execute();
+	}
+	else if (command == "pipe")
+	{
+		auto methodQuery = AstSource::instance().createMethodQuery(node, args);
+		auto toBaseQuery = AstSource::instance().createToClassNodeQuery(node, args);
+		auto compositeQuery = new CompositeQuery();
+		compositeQuery->connectQuery(methodQuery, toBaseQuery);
+		compositeQuery->connectToOutput(toBaseQuery);
+		QueryExecutor queryExecutor(compositeQuery);
+		queryExecutor.execute();
+	}
+	else if (command == "query21")
+	{
+		// Find all classes for which the name contains X and which have a method named Y
+		// 5 queries seems like a lot for this :S
+		auto classesQuery = AstSource::instance().createClassesQuery(node, {"g"});
+		auto filterQuery = new AstNameFilter("Matcher");
+		auto methodsOfQuery = AstSource::instance().createMethodQuery(node, {"of"});
+		auto methodsFilter = new AstNameFilter("matches", true);
```

Comment:
We should talk about these filters and try to find as a generic as possible way of implementing/using them.